File: src/main/java/com/github/unldenis/hologram/line/ITextLine.java
package com.github.unldenis.hologram.line;

import com.github.unldenis.hologram.placeholder.Placeholders;
import org.bukkit.entity.Player;

public interface ITextLine extends ILine<String> {

  boolean isClickable();

  String parse(Player player);

  TextLine asTextLine();

  Placeholders getPlaceholders();

}

File: src/main/kotlin/com/github/unldenis/hologram/line/ITextLine.kt
package com.github.unldenis.hologram.line

import com.github.unldenis.hologram.placeholder.Placeholders
import org.bukkit.entity.Player

interface ITextLine : ILine<String> {

    val clickable : Boolean

    val textLine : TextLine

    val placeholders : Placeholders

    fun parse(player: Player) : String

}